src/http/buf.rs

```diff
@@ -226,6 +240,25 @@ impl MemSlice {
     fn get(&self) -> &[u8] {
         unsafe { &(*self.buf.get())[self.start..self.end] }
     }
+
+    pub fn as_str(&self) -> MemStr {
+        unsafe { MemStr::from_utf8_unchecked(self.clone()) }
+    }
+
+    pub fn find(&self, c: char) -> Option<usize> {
+        let c = c as u8;
+        for (pos, x) in self.get().iter().enumerate() {
+            if *x == c {
+                return Some(pos);
+            }
+        }
+        None
+    }
+
+    #[doc(hidden)]
+    pub fn as_full_str(&self) -> &str {
```

Comment:
It is quite useful actually for some internal operation when you want to translate directly from `MemSlice` to `&str`.